libcloud/compute/drivers/azure_arm.py

```diff
@@ -1811,15 +1951,15 @@ def ex_create_tags(self, resource, tags, replace=False):
             resource = resource.id
         r = self.connection.request(
             resource,
-            params={"api-version": RESOURCE_API_VERSION})
+            params={"api-version": '2018-06-01'})
```

Comment:
Why hardcode the '2018-06-01' here? would this make it incompatible with the '2016-04-30-preview' resource API? Also if any api version is going to be configured it would be nicer to put it in a module level variable at the top of the file like RESOURCE_API_VERSION.

Reply:
I see that there are other api-versions hardcoded in this file already as well, for example [here](https://github.com/apache/libcloud/blob/080b343e0983f2a22f7647b4fd8cebb3339adbe2/libcloud/compute/drivers/azure_arm.py#L1411), so perhaps that is not an issue.

Reply:
@vdloo Thanks for the feedback. I had the same thought that the driver should be using the same version for each API call, but valid objections were raised on an older (now discarded) PR: https://github.com/apache/libcloud/pull/1243 which modified the driver to use a single API version.

And the answer to your question about why I hardcoded the '2018-06-01' is that I was not able to tag a network interface in the 'West US 2' region with the older version.

I discarded the PR which would ensure the driver uses a single API version because it would need to be tested (not just unit testing but also end to end testing) thoroughly. While I have already been using my change in production (only to attach/detach/modify disks and network interfaces) I do not have the resources to test all the functionality of the driver at the moment.